lib/tasks/ldap.rake

```diff
@@ -13,4 +12,17 @@ namespace :ldap do
       end
     end
   end
+  task :remove_deleted_users => :environment do
+    desc <<-END_DESC
+    Deletes all foreman users that authenticate via LDAP, but whose LDAP users do not exist anymore.
+    Also deletes them as the owner of their hosts.
+    END_DESC
+    User.joins(:auth_source).where(:'auth_sources.type' => 'AuthSourceLdap').each do |user|
```

Comment:
I think you need to wrap this in `User.as_anonymous_admin do [...] end` as no user is set when the rake tasks run.